engine/access/state_stream/backend/backend_account_statuses_test.go

```diff
@@ -0,0 +1,582 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/onflow/flow-go/utils/unittest/generator"
+
+	subscriptionmock "github.com/onflow/flow-go/engine/access/subscription/mock"
+
+	"github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/module/blobs"
+	"github.com/onflow/flow-go/module/execution"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data/cache"
+	"github.com/onflow/flow-go/module/mempool/herocache"
+	"github.com/onflow/flow-go/module/metrics"
+	protocolmock "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest/mocks"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine/access/index"
+	"github.com/onflow/flow-go/engine/access/state_stream"
+	"github.com/onflow/flow-go/engine/access/subscription"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+var testProtocolEventTypes = []flow.EventType{
+	state_stream.CoreEventAccountCreated,
+	state_stream.CoreEventAccountContractAdded,
+	state_stream.CoreEventAccountContractUpdated,
+}
+
+// Define the test type struct
+type testType struct {
```

Comment:
```suggestion
type testCase struct {
```

Reply:
As in other tests related to backend subscriptions(`SubscribeEventsSuite`, `BackendBlocksSuite`, `BackendEventsSuite`) being used  `testType struct ` so I decided to keep the same naming.